File: benchmarks/cross-language-v1/tasks/stable-dispatch-order-v1/public/typescript/candidate.ts
export function dispatchOrder(aPriority: number, bPriority: number, cPriority: number): number {
  if (aPriority <= bPriority && aPriority <= cPriority) {
    return bPriority <= cPriority ? 123 : 132;
  }
  if (bPriority <= aPriority && bPriority <= cPriority) {
    return aPriority <= cPriority ? 213 : 231;
  }
  return aPriority <= bPriority ? 312 : 321;
}

File: benchmarks/cross-language-v1/tasks/stable-dispatch-order-v1/public/rust/candidate.rs
pub fn dispatch_order(a_priority: i64, b_priority: i64, c_priority: i64) -> i64 {
    if a_priority <= b_priority && a_priority <= c_priority {
        if b_priority <= c_priority {
            123
        } else {
            132
        }
    } else if b_priority <= a_priority && b_priority <= c_priority {
        if a_priority <= c_priority {
            213
        } else {
            231
        }
    } else if a_priority <= b_priority {
        312
    } else {
        321
    }
}
